### PR TITLE
Fix operator ServiceMonitor not being created for controller metrics

### PR DESCRIPTION
**Pull request by IshwarKanse**
**Description:** <Describe what has changed.>
  This PR fixes a bug where the OpenTelemetry Operator fails to create a ServiceMonitor for its own metrics on OpenShift clusters.

  Two issues were identified and fixed:

  1. Missing operator args in OpenShift deployment: PR #4576 (Remove kube-rbac-proxy) inadvertently removed critical operator flags from the OpenShift deployment. The manager_auth_proxy_tls_patch.yaml is a strategic merge patch that replaces the entire args array, overwriting the args set by manager-patch.yaml. This removed flags like --create-sm-operator-metrics=true, --enable-leader-election, --enable-cr-metrics=true, etc.

Fix: Move TLS cert args (--metrics-tls-cert-file, --metrics-tls-key-file) to manager-patch.yaml and remove the args section from manager_auth_proxy_tls_patch.yaml.

  2. Invalid scheme value rejected by CRD validation: PR #3858 (Bump prometheus group) updated the prometheus-operator dependency where the SchemeHTTPS constant is uppercase "HTTPS". The ServiceMonitor CRD validation only accepts lowercase values, causing creation to fail with:
  spec.endpoints[0].scheme: Unsupported value: "HTTPS": supported values: "http", "https"

Fix: Change ptr.To(monitoringv1.SchemeHTTPS) to ptr.To(monitoringv1.Scheme("https")) in internal/operator-metrics/metrics.go.

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #4603
- https://issues.redhat.com/browse/TRACING-5919 

**Testing:** <Describe what testing was performed and which tests were added.>
  Tested on OpenShift cluster by running the e2e-openshift monitoring test:
  chainsaw test --skip-delete tests/e2e-openshift/monitoring

  The test verifies that:
  - Operator ServiceMonitor is created in the operator namespace
  - Operator controller metrics (controller_runtime_reconcile_time_seconds_count{controller="opentelemetrycollector"}) are available in Thanos/Prometheus

**Documentation:** <Describe the documentation added.>
No documentation changes required.

### CODE REVIEW

File: internal/operator-metrics/metrics.go [hunk 1]
```diff
@@ -166,7 +166,7 @@ func (om operatorMetrics) createOperatorMetricsServiceMonitor(ctx context.Contex
 					BearerTokenFile: bearerTokenFile,
 					Interval:        "30s",
 					Path:            "/metrics",
-					Scheme:          ptr.To(monitoringv1.SchemeHTTPS),
+					Scheme:          ptr.To(monitoringv1.Scheme("https")),
```
Comment (swiatekm) [inline internal/operator-metrics/metrics.go:169]: This is kind of nasty. The validation in the CRD version we depend on actually accepts the uppercase version as well, but older versions do not. Technically, we're misusing the type here, but I don't know if there's another way. I'm surprised this doesn't work as-is considering monitoringv1.Scheme serializes to the lowercase version.

Relatedly, I'm surprised we don't have integration tests for this. Or do we, and we always use newer prometheus-operator CRDs?

Comment (IshwarKanse) [reply internal/operator-metrics/metrics.go:169]: @swiatekm   The issue is that CRD validation happens before serialization. When we pass monitoringv1.SchemeHTTPS (which is the string "HTTPS"), the Kubernetes API server validates the value against the CRD's enum constraint before the object is created.

  The prometheus-operator library defines:
  const SchemeHTTPS URIScheme = "HTTPS"

  With validation annotation:
  // +kubebuilder:validation:Enum=http;https;HTTP;HTTPS

  Newer CRD versions accept both cases, but older versions only accept lowercase http and https. Since we need to support all versions, using the lowercase string directly seems to be the pragmatic solution.
  
  For the test, I have updated the operator-metrics test to enable the creation of controller manager metrics service monitor and assert it.

Comment (swiatekm) [reply internal/operator-metrics/metrics.go:169]: We're defining the ServiceMonitor as a Go struct, but it gets serialized to JSON before it's submitted to the API Server. And the Scheme type has a `String()` method that returns it in lowercase. So my assumption was that it would get serialized to JSON as lowercase too, but this is clearly not the case, and I'm not sure why.

It's good that you added that test, but even if we had it, it wouldn't have caught this problem, because for e2e tests we install prometheus-operator CRDs straight from their main branch. We should probably use the ones from the version we import, but that still wouldn't have caught the problem, as it only appears on older versions. I don't know that we can do anything about it, it's just nasty.

Comment (IshwarKanse) [reply internal/operator-metrics/metrics.go:169]: Agreed, this is difficult to catch in upstream, but we were able to catch this issue using the OpenShift-specific test case at tests/e2e-openshift/monitoring, which runs against actual OpenShift clusters with their shipped CRD versions. Although there is some gap between when changes are merged upstream and when our OpenShift CI jobs pick up those changes and run the tests, it still provides a safety net for these platform-specific edge cases.
